isis/tests/FunctionalTestsMsi2isis.cpp

```diff
@@ -0,0 +1,469 @@
+  }
+  Cube cube(cubeFileName);
+  Pvl *isisLabel = cube.label();
+
+
+  // Dimensions Group
+  ASSERT_EQ(cube.sampleCount(), 537);
+  ASSERT_EQ(cube.lineCount(), 412);
+  ASSERT_EQ(cube.bandCount(), 1);
+
+  // Instrument Group
+  PvlGroup &inst = isisLabel->findGroup("Instrument", Pvl::Traverse);
+  ASSERT_EQ(inst["SpacecraftName"][0].toStdString(), "NEAR EARTH ASTEROID RENDEZVOUS");
+  ASSERT_EQ(inst["InstrumentId"][0].toStdString(), "MSI" );
+  ASSERT_EQ(inst["TargetName"][0].toStdString(), "EROS" );
+  ASSERT_EQ(inst["SpacecraftClockStartCount"][0].toStdString(), "126865998830" );
+  ASSERT_EQ(inst["SpacecraftClockStopCount"][0].toStdString(), "126865998919" );
+  ASSERT_EQ(inst["OriginalSpacecraftClockStartCount"][0].toStdString(), "126865998.830" );
+  ASSERT_EQ(inst["OriginalSpacecraftClockStopCount"][0].toStdString(), "126865998.919" );
+  ASSERT_DOUBLE_EQ(double(inst["ExposureDuration"]), 89.0);
+  ASSERT_EQ(inst["StartTime"][0].toStdString(), "2000-02-25T05:16:12.656" );
+  ASSERT_EQ(inst["StopTime"][0].toStdString(), "2000-02-25T05:16:12.745" );
+  ASSERT_DOUBLE_EQ(double(inst["DpuDeckTemperature"]), 286.5);
+  ASSERT_EQ(inst["DpuDeckTemperature"].unit().toStdString(), "K");
+
+  // Archive Group
+  PvlGroup &archive = isisLabel->findGroup("Archive", Pvl::Traverse);
+  ASSERT_EQ(archive["DataSetId"][0].toStdString(), "NEAR-A-MSI-3-EDR-EROS/ORBIT-V1.0");
+  ASSERT_EQ(archive["ProductId"][0].toStdString(), "M0126865998F4_2P_IOF.FIT");
+  ASSERT_EQ(archive["InstrumentHostName"][0].toStdString(), "NEAR EARTH ASTEROID RENDEZVOUS");
+  ASSERT_EQ(archive["InstrumentName"][0].toStdString(), "MULTI-SPECTRAL IMAGER");
+  ASSERT_EQ(archive["InstrumentId"][0].toStdString(), "MSI");
+  ASSERT_EQ(archive["TargetName"][0].toStdString(), "EROS");
+
+  // BandBin Group
+  PvlGroup &bandbin = isisLabel->findGroup("BandBin", Pvl::Traverse);
+  ASSERT_EQ(int(bandbin["FilterNumber"]), 4);
+  ASSERT_EQ(int(bandbin["Center"]), 950);
+  ASSERT_EQ(bandbin["Center"].unit().toStdString(), "nm");
+
+ // Kernels Group
+  PvlGroup &kernel = isisLabel->findGroup("Kernels", Pvl::Traverse);
+  ASSERT_EQ(int(kernel["NaifFrameCode"]), -93001);
+
+  std::unique_ptr<Histogram> hist (cube.histogram());
+
+  ASSERT_NEAR(hist->Average(), 0.009091271356228, .000001);
+  ASSERT_NEAR(hist->Sum(), 1481.568127839, .0001);
+  ASSERT_EQ(hist->ValidPixels(), 162966);
+  ASSERT_EQ(hist->NullPixels(), 58278);
+  ASSERT_NEAR(hist->StandardDeviation(), 0.013542099831273, .0001);
+}
+
+
+// Error tests
+TEST(Msi2isis, Msi2isisTestAMissingLabelFile) {
+   Pvl appLog;
+   QTemporaryDir prefix;
+   QString dAPP_XML = FileName("$ISISROOT/bin/xml/msi2isis.xml").expanded();
+   QString cubeFileName = prefix.path() + "/image.cub";
+   QVector<QString> args = {"from=data/near/msi2isis/image.fit",
+                            "to=" + cubeFileName };
+
+   UserInterface options(dAPP_XML, args);
+
+   // TEST A: pass in a fits file without label
+   ASSERT_ANY_THROW(msi2isis(options, &appLog));
```

Comment:
We use a more complex method to test exceptions that checks the message and error type is what we expect.

see https://github.com/USGS-Astrogeology/ISIS3/wiki/Writing-ISIS3-Tests-Using-Gtest-and-Ctest#testing-exceptions

You can also use an `EXPECT_THAT` to check the substring like this https://github.com/USGS-Astrogeology/ISIS3/blob/dev/isis/tests/FunctionalTestsApolloremrx.cpp#L119

Reply:
Done.